shared/domains/chelonia/chelonia.js

```diff
@@ -348,26 +484,41 @@ export default (sbp('sbp/selectors/register', {
       return events.reverse().map(b64ToStr)
     }
   },
+  'chelonia/currentContractState': function (contractID: string) {
+    if (sbp(this.config.stateSelector)[contractID]) {
+      return cloneDeep(sbp(this.config.stateSelector)[contractID])
+    }
+  },
   'chelonia/latestContractState': async function (contractID: string) {
-    const events = await sbp('chelonia/out/eventsSince', contractID, contractID)
-    let state = {}
+    const rootState = sbp(this.config.stateSelector)
+    if (rootState[contractID] && Object.keys(rootState[contractID]).some((x) => x !== '_volatile')) {
```

Comment:
What is this second `Object.keys(rootState[contractID]).some((x) => x !== '_volatile')` condition for?

Reply:
This is checking whether `rootState` has a key _other than_ `_volatile`. If this is the case, the object is cloned (otherwise, if it only has `_volatile`, it can be treated as if it wasn't set).